src/Components/SceneRefreshButton/SceneRefreshButton.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+import {
+    ISceneRefreshButtonProps,
+    ISceneRefreshButtonStyleProps,
+    ISceneRefreshButtonStyles
+} from './SceneRefreshButton.types';
+import {
+    ANIMATION_DURATION_SECONDS,
+    getStyles
+} from './SceneRefreshButton.styles';
+import {
+    classNamesFunction,
+    useTheme,
+    styled,
+    TooltipHost,
+    DirectionalHint,
+    Stack
+} from '@fluentui/react';
+import { useId } from '@fluentui/react-hooks';
+import HeaderControlButton from '../HeaderControlButton/HeaderControlButton';
+import HeaderControlGroup from '../HeaderControlGroup/HeaderControlGroup';
+import { formatTimeInRelevantUnits } from '../../Models/Services/Utils';
+import { useTranslation } from 'react-i18next';
+import { DurationUnits } from '../../Models/Constants';
+
+const getClassNames = classNamesFunction<
+    ISceneRefreshButtonStyleProps,
+    ISceneRefreshButtonStyles
+>();
+
+const ROOT_LOC = 'sceneRefreshButton';
+const LOC_KEYS = {
+    lastRefreshed: `${ROOT_LOC}.lastRefreshed`,
+    refreshRate: `${ROOT_LOC}.refreshRate`
+};
+const SceneRefreshButton: React.FC<ISceneRefreshButtonProps> = (props) => {
+    const {
+        isRefreshing,
+        lastRefreshTimeInMs,
+        onClick,
+        refreshFrequency,
+        styles
+    } = props;
+    const iconAnimationTimeout = useRef<NodeJS.Timeout>();
+
+    // state
+    const [isRefreshInProgress, setIsRefreshInProgress] = useState<boolean>(
+        false
+    );
+    const [timeSinceLastRefresh, setTimeSinceLastRefresh] = useState<{
+        value: number;
+        displayStringKey: string;
+    }>({
+        value: 0,
+        displayStringKey: 'duration.seconds'
+    });
+
+    // hooks
+    const buttonId = useId();
+    const { t } = useTranslation();
+
+    // styles
+    const classNames = getClassNames(styles, {
+        isRefreshing: isRefreshInProgress,
+        theme: useTheme()
+    });
+
+    // set local state when we are notified about refreshing, then give it time to animate
+    useEffect(() => {
+        if (isRefreshing) {
+            setIsRefreshInProgress(true); // apply the styling
+            clearTimeout(iconAnimationTimeout.current); // clear any pending timeouts
+            setTimeout(() => {
+                setIsRefreshInProgress(false);
+            }, ANIMATION_DURATION_SECONDS * 1000 + 0.1); // give it enough time to finish the animation, then remove the style
+        }
+    }, [isRefreshing]);
+
+    // to get live updating we have to trigger renders and recalculate on a regular cadence so set a timer to keep checking
+    setTimeout(() => {
+        const timeSince =
+            lastRefreshTimeInMs > 0 ? Date.now() - lastRefreshTimeInMs : 0;
```

Comment:
lastRefreshTimeInMs is a number type

Reply:
Date.now() is also a number, the milliseconds since Jan 1, 1970. If `lastRefreshTimeInMs` is calculated with Date it will never be 0.

Reply:
it's just a safety net if it doesn't get the prop. It won't ever actually execute. I probably added it when my story blew up up for this control because I wasn't passing the prop correctly.